pkg/virt-operator/resource/generate/components/util.go

```diff
@@ -0,0 +1,13 @@
+package components
+
+import "fmt"
+
+func GeneratePromQlQueryForValueDifferenceInHourInterval(metric string) string {
+	return GeneratePromQlQueryForValueDifferenceInInterval(metric, 60)
+}
+func GeneratePromQlQueryForValueDifferenceInFiveMinutesInterval(metric string) string {
+	return GeneratePromQlQueryForValueDifferenceInInterval(metric, 5)
+}
+func GeneratePromQlQueryForValueDifferenceInInterval(metric string, timeIntervalInMinutes int) string {
+	return fmt.Sprintf("clamp_min(sum (%s ) - sum (%s offset %dm) , 0  )", metric, metric, timeIntervalInMinutes)
```

Comment:
hm, I think it can be simplified a bit. This expression shows the same numbers:

`clamp_min(sum(rest_client_requests_total{pod=~'virt-controller-.*'} - rest_client_requests_total{pod=~'virt-controller-.*'} offset 5m), 0)`

so in your case this might work:
`clamp_min(sum(%s - %s offset %dm), 0)`

Reply:
There is a problem with this query because if we will get new kind of request in 5 minutes interval it won't be included,
For instance:

**time x** :  lets say we had the following requests vector:   {metricA 2, metricB 1}

and after 2 minutes we get new kind of request we can see it in metricC

so in **time x+5m** we will have the following metrics: {metricA 2 , metricB 1, **metricC 1**}

if we will use 'clamp_min(sum(%s - %s offset %dm), 0)' instead :

`sum( {metricA 2, metricB 1, metricC 1} - {metricA 2 , metricB 1}) = 0 `   **instead** of 1 which is the real result 

this is explained in the following link:
https://prometheus.io/docs/prometheus/latest/querying/operators/#arithmetic-binary-operators

"Between two instant vectors, a binary arithmetic operator is applied to each entry in the left-hand side vector and its matching element in the right-hand vector. The result is propagated into the result vector with the grouping labels becoming the output label set. The metric name is dropped. **Entries for which no matching entry in the right-hand vector can be found are not part of the result.**"